src/sentry/models/relay.py

```diff
@@ -31,5 +31,21 @@ def has_org_access(self, org):
         # Internal relays always have access
         if self.is_internal:
             return True
+
+        trusted_relays = org.get_option("sentry:trusted-relays", [])
+        trusted_keys = [
+            key
+            for relay_info in trusted_relays
+            if relay_info is not None
+            for key in [relay_info.get("public_key")]
+            if key is not None and len(key) > 0
+        ]
         # Use the normalized form of the public key for the check
-        return six.text_type(self.public_key_object) in org.get_option("sentry:trusted-relays", [])
+        return six.text_type(self.public_key_object) in trusted_keys
```

Comment:
Can you rewrite this into either a simple for loop or a call to `any()`? At this point, the stored key should already be validated (non-empty). For example:

```py
        key = six.text_type(self.public_key_object)
        for trusted_relay in org.get_option("sentry:trusted-relays", []):
            if trusted_relay and trusted_relay.get("public_key") == key
                return True

        return False
```

Reply:
True, it should already be validated but it never hurts to be sure, to be sure :-) , especially since this key is reused and it contained strings before. 

Any way I agree that that comprehension is not very comprehensible :-) . 

Changed it to a filter

```
key = six.text_type(self.public_key_object)
keys = filter(lambda info: info is not None and info.get(u"public_key") == key,
              trusted_relays)

return len(keys) > 0
```